src/tests/test_connection.py

```diff
@@ -266,3 +267,31 @@ def test_no_current_connection_and_get_info(monkeypatch, mock_database):
 
     monkeypatch.setattr(conn, "session", None)
     assert conn._get_curr_sqlalchemy_connection_info() is None
+
+
+@pytest.mark.parametrize(
+    "conn, expected",
+    [
+        [sqlite3.connect(""), True],
+        [
+            CustomConnection(engine=sqlalchemy.create_engine("sqlite://")),
+            True,
+        ],
+        [
+            Connection(engine=sqlalchemy.create_engine("sqlite://")),
+            False,
+        ],
+        ["not_a_valid_connection", False],
+        [0, False],
+    ],
+    ids=[
+        "sqlite3_connection",
+        "custom_connection",
+        "normal_connection",
+        "str",
+        "int",
+    ],
+)
+def test_custom_connection(conn, expected):
```

Comment:
add one extra test case, a mock object that has some (but not all) properties required in PEP 249

Reply:
Added